src/gatsby/models/authors/parse-link.js

```diff
@@ -0,0 +1,94 @@
+/*
+   Processor builder: makes a function that turns a string into processed Link metadata
+
+   Takes "Mod functions" as parameters, returns a function that starts with {} and pipes
+   that through each function, then returns the result.
+
+   signature: ({ scheme: string, host: string, pathname: string}, input)
+   example:      ^ https://      ^ mysite.com  ^ /rest/of/the/url ^ an object
+
+   The second parameter will be the current state of the input, and usually will
+   have to be spread into your return result.
+ */
+function processor(...mods) {
+  return function process(groups) {
+    return mods.reduce(
+      (currentResult, _, i) => mods[i](groups, currentResult),
+      {}
+    )
+  }
+}
+
+// Processor helpers
+
+function trimSlashes(input) {
+  return /^\/?(.*?)\/?$/.exec(input)[1]
+}
+
+const asSite = site => (groups, input) => ({
+  ...input,
+  site
+})
+
+const pathnameAsUsername = ({ pathname }, input) => ({
+  ...input,
+  username: trimSlashes(pathname)
+})
+
+const urlHTTPS = ({ host, pathname }, input) => ({
+  ...input,
+  url: 'https://' + host + pathname
+})
+
+const urlDefault = ({ scheme, host, pathname }, input) => ({
+  ...input,
+  url: (scheme || 'https://') + host + pathname
+})
+
+// Building processors for recognized sites
+
+const processors = {
+  'twitter.com': processor(asSite('twitter'), urlHTTPS, pathnameAsUsername),
+  'github.com': processor(asSite('github'), urlHTTPS, pathnameAsUsername),
+  'linkedin.com': processor(
```

Comment:
for linkedin it's better to include www? www.linkedin.com is canonical

Reply:
Can do!

Reply:
Done! LinkedIn now specifically adds `www` to match the canonical link.